File: repos_estudiantes_ALVAREZ_segunda_entrega/AGUILAR_MACHICADO_NILS_REINER/PRACTICA 2/COMPOSICION-AGREGACION/Ejercicio3.py
class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso

    def mostrar_info(self):
        return f"Parte: {self.nombre}, Peso: {self.peso} kg"

# SIN GETTERS NI SETTERS

class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = []

    def agregar_parte(self, parte):
        self.partes.append(parte)

    def mostrar_avion(self):
        print(f"Avión modelo: {self.modelo}, Fabricante: {self.fabricante}")
        for p in self.partes:
            print(p.mostrar_info())

if __name__ == "__main__":
    avion = Avion("Boeing 747", "Boeing Co.")
    avion.agregar_parte(Parte("Motor", 3000))
    avion.agregar_parte(Parte("Ala Izquierda", 1500))
    avion.agregar_parte(Parte("Tren de Aterrizaje", 800))
    avion.mostrar_avion()

File: repos_estudiantes_ALVAREZ_segunda_entrega/AGUILAR_MACHICADO_NILS_REINER/PRACTICA 2/COMPOSICION-AGREGACION/Ejercicio3.java
import java.util.ArrayList;

class Parte {
    private String nombre;
    private double peso;

    public Parte(String nombre, double peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

    public String mostrarInfo() {
        return "Parte: " + nombre + ", Peso: " + peso + " kg";
    }
}

class Avion {
    private String modelo;
    private String fabricante;
    private ArrayList<Parte> partes;

    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
    }

    public void agregarParte(Parte parte) {
        partes.add(parte);
    }

    public void mostrarAvion() {
        System.out.println("Avión modelo: " + modelo + ", Fabricante: " + fabricante);
        for (Parte p : partes) {
            System.out.println(p.mostrarInfo());
        }
    }
}
// SIN GETTERS NI SETTERS
class TestAvion {
    public static void main(String[] args) {
        Avion avion = new Avion("Boeing 747", "Boeing Co.");
        avion.agregarParte(new Parte("Motor", 3000));
        avion.agregarParte(new Parte("Ala Izquierda", 1500));
        avion.agregarParte(new Parte("Tren de Aterrizaje", 800));

        avion.mostrarAvion();
    }
}